modules/apps/analytics/analytics-cms-rest-impl/src/main/java/com/liferay/analytics/cms/rest/internal/resource/v1_0/OverviewContentResourceImpl.java

```diff
@@ -19,4 +38,217 @@
 )
 public class OverviewContentResourceImpl
 	extends BaseOverviewContentResourceImpl {
+
+	@Override
+	public OverviewContent getOverviewContent(
+			String languageId, Integer rangeKey, Integer spaceId)
+		throws Exception {
+
+		return _toOverviewContent(
```

Comment:
@rachaelkoestartyo can you please send a follow up checking the permission of `spaceId` and including it into the query?
Please see [DepotEntryServiceImpl](https://github.com/liferay/liferay-portal/blob/319f4541315aa345aed334837345be457df28c75/modules/apps/depot/depot-service/src/main/java/com/liferay/depot/service/impl/DepotEntryServiceImpl.java#L90) and [AssetMetricResourceImpl](https://github.com/liferay/liferay-portal/blob/master/modules/dxp/apps/analytics/analytics-reports-rest-impl/src/main/java/com/liferay/analytics/reports/rest/internal/resource/v1_0/AssetMetricResourceImpl.java) as references.
Bear in mind that if `spaceId` is `null`, it should return the metrics for all the spaces the user has permission to view.

Reply:
hi @marcosapmf 
I've been looking into this, one thing that i think i need in the case when spaceId is null is for `DepotEntryServiceImpl` to have a method to get all depotEntries that the contextUser has permission to view. This method isn't available currently. Can i add it to DepotEntryServiceImpl? or do i just use the local service DepotEntryLocalServiceImpl#getDepotEntries and then for each returned depotEntry, check against permission checker?

Reply:
Hi @rachaelkoestartyo,
I think you can create the method in `DepotEntryServiceImpl`. Another option is trying to do as they do on their own [endpoint](https://github.com/liferay/liferay-portal/blob/3aa404569c09f52ef69627bdb2c16d5edc3cc1da/modules/apps/headless/headless-asset-library/headless-asset-library-impl/src/main/java/com/liferay/headless/asset/library/internal/resource/v1_0/AssetLibraryResourceImpl.java#L84), getting the ids from the index and check the permission.
I think the second option has a better performance than getting the the DepotEntries directly from the database.